src/main/java/com/songoda/epiclevels/database/DataUpdaterAbstract.java

```diff
@@ -0,0 +1,128 @@
+package com.songoda.epiclevels.database;
+
+import com.songoda.core.database.DatabaseConnector;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+import java.util.function.Consumer;
+
+public abstract class DataUpdaterAbstract {
+
+    private final ReadWriteLock lock = new ReentrantReadWriteLock();
+    private long currentID = -1;
+    private boolean enabled = false;
+
+    public void onEnable() {
+        getConnector().connect(connection -> {
+            // Taken from LuckPerms
+            String createTable = "CREATE TABLE IF NOT EXISTS `" + getTablePrefix() + "messenger` (`id` INT AUTO_INCREMENT NOT NULL, `time` TIMESTAMP NOT NULL, `msg` TEXT NOT NULL, PRIMARY KEY (`id`)) DEFAULT CHARSET = utf8mb4";
+            try (Statement statement = connection.createStatement()) {
+                try {
+                    statement.execute(createTable);
+                } catch (SQLException e) {
+                    if (e.getMessage().contains("Unknown character set")) {
+                        statement.execute(createTable.replace("utf8mb4", "utf8"));
+                    } else {
+                        throw e;
+                    }
+                }
+            }
+
+            try (PreparedStatement statement = connection.prepareStatement("SELECT MAX(`id`) as `latest` FROM `" + getTablePrefix() + "messenger`")) {
+                try (ResultSet resultSet = statement.executeQuery()) {
+                    if (resultSet.next()) {
+                        currentID = resultSet.getLong("latest");
+                    }
+                }
+            }
+            enabled = true;
+        });
+    }
+
+    public void onDisable() {
+        enabled = false;
+    }
+
+    public abstract DatabaseConnector getConnector();
+
+    public abstract String getTablePrefix();
+
+    public boolean isEnabled() {
+        return enabled;
+    }
+
+    public void setEnabled(boolean enabled) {
+        this.enabled = enabled;
+    }
+
+    public void sendMessage(String msg) {
+        lock.readLock().lock();
```

Comment:
`#isEnabled()` just returns a boolean, could we avoid the lock and return instantly if it returns `false`?
(As `#setEnabled(boolean)` does not lock itself I don't see any benefit.